api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -310,3 +310,26 @@ message Principal {
     Principal not_id = 8;
   }
 }
+
+// Action defines the result of allowance or denial when a request matches the matcher.
+message Action {
+  // The name indicates the matcher name.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
This feels redundant from the API perspective. There's an outer `name` field for the entire action proto. Is there a difference between the two?

Reply:
The outer `name` is the name of typed config and the inner `name` is the real name of the matched path. As you know, we cannot get the typed config's name, it is just a hint for the factory to find the correct action.

Reply:
The outer `name` is an action name, and is no longer used by the factory since the look-up is primarily based on the type URL. It can be used by ECDS as a "resource name" but that seems OK as well. The purpose of typed_config name is semantic now, so it seems like it is reasonable to re-use. The implementation interface will need to be updated to propagate the name next to the proto.

@markdroth might have some thoughts here.

Reply:
I have checked the code path, and found that `envoy.reloadable_features.no_extension_lookup_by_name` is still a runtime feature. Users can still turn off the feature and let Envoy parse typed config by its name. I think currently we cannot reuse the `name` in typed config.

Reply:
@zhxie The name in the extension config is definitely meant to be semantic since that's the only way to support repeated filters (for ECDS, for per filter route config). Even with this runtime flag on, the lookup would first use the typed config and only fallback to using name when that doesn't work: https://github.com/envoyproxy/envoy/blob/cf840b8c4822a692184e3df471df89e4973e72fb/source/common/config/utility.h#L295.

So I think you can take the liberty to require type lookup in this new API. In practice, most users will probably stick the same name in both places anyways.

Reply:
It is fine to use the outer name, but there is one another problem. `Envoy::Matcher::ActionFactory::createActionFactoryCb` takes the inner typed config as its parameter, and we cannot get its outer name unless we change the interface.

Reply:
OK, let's leave the redundant name since it isn't the only instance. I've tried removing the outer name, but it's marked as required (see https://github.com/cncf/xds/pull/28). We can clean up the verbosity later.